test/visible-on-ancestor-behavior.html

```diff
@@ -0,0 +1,201 @@
+<!doctype html>
+<html>
+	<head>
+		<meta charset="UTF-8">
+		<title>d2l-visible-on-ancestor-behavior tests</title>
+		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">
+		<script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+		<script src="../../web-component-tester/browser.js"></script>
+		<link rel="import" href="../../polymer/polymer.html">
+		<link rel="import" href="../d2l-focusable-behavior.html">
+		<link rel="import" href="../d2l-visible-on-ancestor-behavior.html">
+	</head>
+	<body>
+		<dom-module id="d2l-visible-on-ancestor-behavior-test">
+			<template>
+				<button class="d2l-focusable"><slot></slot></button>
+			</template>
+			<script>
+				Polymer({
+					is: 'd2l-visible-on-ancestor-behavior-test',
+					behaviors: [
+						D2L.PolymerBehaviors.FocusableBehavior,
+						D2L.PolymerBehaviors.VisibleOnAncestorBehavior
+					]
+				});
+			</script>
+		</dom-module>
+
+		<test-fixture id="noVisibleOnAncestor">
+			<template>
+				<div class="d2l-visible-on-ancestor-target">
+					<d2l-visible-on-ancestor-behavior-test id="wc1">focusable 1</d2l-visible-on-ancestor-behavior-test>
+				</div>
+			</template>
+		</test-fixture>
+
+		<test-fixture id="noVisibleOnAncestorTarget">
+			<template>
+				<div>
+					<d2l-visible-on-ancestor-behavior-test id="wc1" visible-on-ancestor>focusable 1</d2l-visible-on-ancestor-behavior-test>
+				</div>
+			</template>
+		</test-fixture>
+
+		<test-fixture id="simpleVisibleOnAncestor">
+			<template>
+				<div>
+					<d2l-visible-on-ancestor-behavior-test id="wc1">focusable 1</d2l-visible-on-ancestor-behavior-test>
+					<div class="d2l-visible-on-ancestor-target" tabindex="0">
+						<d2l-visible-on-ancestor-behavior-test id="wc2" tabindex="0">focusable 2</d2l-visible-on-ancestor-behavior-test>
+						<d2l-visible-on-ancestor-behavior-test id="wc3" visible-on-ancestor>focusable 3</d2l-visible-on-ancestor-behavior-test>
+					</div>
+				</div>
+			</template>
+		</test-fixture>
+
+		<script>
+
+		describe('d2l-visible-on-ancestor-behavior', function() {
+
+			var voaFixture;
+
+			var isDisplayed = function(node) {
+				return !node.hasAttribute('d2l-visible-on-ancestor-hide');
+			};
+
+			var dispatchMouseEvent = function(node, name) {
+				var e = new MouseEvent(name, {
+					bubbles: false,
+					cancelable: false,
+					clientX: 0,
+					clientY: 0,
+					composed: true,
+					buttons: 0
+				});
+				node.dispatchEvent(e);
+			};
+
+			describe('no visible-on-ancestor elements', function() {
+
+				beforeEach(function(done) {
+					voaFixture = fixture('noVisibleOnAncestor');
+					requestAnimationFrame(function() {
```

Comment:
I think in my previous attempt I have the `fixture` call inside of the `requestAnimationFrame`, which obviously defeats the purpose.  This seems to work ok.